inference-engine/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp

```diff
@@ -176,6 +176,217 @@ TEST_P(OVInferRequestIOTensorTest, canInferWithGetOut) {
+
+    const std::string tensor_name = "input_tensor";
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[tensor_name] = {{1}, {1, 10}, {10, 12}, {10, 12}, {8}, {16}};
+    OV_ASSERT_NO_THROW(func->reshape(shapes));
+    OV_ASSERT_NO_THROW(execNet = core->compile_model(func, targetDevice, configuration));
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+
+    float data[51200];
+    auto el = ngraph::element::Type_t::f32;
+
+    auto byteStrides = [](const ov::Strides& strides, const ov::element::Type& type) -> ov::Strides {
+        ov::Strides byte_strides(strides.size());
+        for (size_t i = 0; i < strides.size(); ++i)
+            byte_strides[i] = strides[i] * type.size();
+        return byte_strides;
+    };
+
+    ov::runtime::Tensor tensor1(el, {1, 4, 10, 10, 8, 16}, data, byteStrides({51200, 12800, 1280, 128, 16, 1}, el));
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", tensor1));
+
+    ov::runtime::Tensor actualTensor = req.get_tensor("ReLU");
+    OV_ASSERT_NO_THROW(actualTensor.set_shape({1, 4, 20, 20}));
+
+    OV_ASSERT_NO_THROW(req.infer());
+
+    ov::runtime::Tensor outTensor = req.get_tensor("ReLU");
+    ASSERT_EQ(actualTensor.get_shape(), outTensor.get_shape());
+    for (int i = 0; i < actualTensor.get_size(); i++) {
+        if (actualTensor.data<float>()[i] >= 0)
+            ASSERT_EQ(actualTensor.data<float>()[i], outTensor.data<float>()[i]);
+        else
+            ASSERT_EQ(0, outTensor.data<float>()[i]);
+    }
+
+    ov::runtime::Tensor tensor2(el, {1, 10, 10, 8, 16, 4}, data, byteStrides({51200, 5120, 512, 64, 4, 1}, el));
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+    ASSERT_THROW(req.set_tensor("input_tensor", tensor2), ov::Exception);
+// [Discuss] Identical tensor in case:
+// ie::BlockingDesc blockingDesc1{{1, 4, 10, 10, 8, 16}, {0, 1, 2, 3, 4, 5}};
+// ov::runtime::Tensor tensor1(fp32, blockingDesc1.getBlockDims(), data, byteStrides(blockingDesc1.getStrides(), el));
+// ie::BlockingDesc blockingDesc2{{1, 4, 10, 10, 8, 16}, {0, 2, 3, 4, 5, 1}};
+// ov::runtime::Tensor tensor2(fp32, blockingDesc2.getBlockDims(), data, byteStrides(blockingDesc2.getStrides(), el));
+}
+
+TEST_P(OVInferRequestIOTensorSetTensorGetTensorDynamicShapesTest, specialCase3) {
+    const std::string tensor_name = "input_tensor";
+    std::map<std::string, ov::PartialShape> shapes;
+    shapes[tensor_name] = {{1, 5}, {1, 5}, {10, 20}, {10, 20}};
+    OV_ASSERT_NO_THROW(function->reshape(shapes));
+    OV_ASSERT_NO_THROW(execNet = core->compile_model(function, targetDevice, configuration));
+    OV_ASSERT_NO_THROW(req = execNet.create_infer_request());
+
+    ov::runtime::Tensor actualTensor = ov::test::utils::create_and_fill_tensor(element::f32, {1, 4, 20, 20}, 10, 1);
+    OV_ASSERT_NO_THROW(req.set_tensor("input_tensor", actualTensor));
+    ov::runtime::Tensor outTensor = req.get_tensor("ReLU");
+    OV_ASSERT_NO_THROW(outTensor.set_shape({1, 4, 20, 20}));
```

Comment:
I just understood that for output tensor we need test some different scenario
1. We need to set tensor with preallocated memory and check that after inference we store correct values in buffer
2. Call `set_tensor` for output

Reply:
why do we need set_shape for outTensor ?

Reply:
I guess we need to test all possible scenario